app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java

```diff
@@ -1,66 +1,273 @@
+    }
+
+    @Override
+    protected void populateList(Query query) {
+        //get all of the habits
+        Task<QuerySnapshot> querySnapshotTask = query.get();
+
+        /*
+        populate HabitList with current Habits and habit IDs to initialize state to match
+        database, fills when habitList is empty and snapshot is not, which is only
+        when app is initially launched
+        */
+        if (_habitList.getHabitList().isEmpty()) {
+            // wait for all the snapshots to come in
+            while (!querySnapshotTask.isComplete()) ;
+
+            // make a list of all of the habit snapshots
+            List<DocumentSnapshot> snapshotList = querySnapshotTask.getResult().getDocuments();
+
+            // convert all of the snapshots into proper habits
+            for (int i = 0; i < snapshotList.size(); i++) {
+                // get the data and convert to hashmap
+                Map<String, Object> extractMap = snapshotList.get(i).getData();
+
+                // create a new habit with the snapshot data
+                Habit addHabit = makeHabit(extractMap);
+
+                // add the habit to the local list
+                _habitList.addHabitLocal(addHabit);
+
+                Log.d("Added Habit", _habitList.getHabit(i).getTitle());
+
+
+            }
+        }
+    }
+
+    @Override
+    protected void initializeRecyclerView(LinearLayoutManager layoutManager, View view) {
+        RecyclerView recyclerView = view.findViewById(R.id.home_recycler_view);
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setLayoutManager(layoutManager);
+        recyclerView.addItemDecoration(new DividerItemDecoration(getContext(), LinearLayoutManager.VERTICAL));
+        recyclerView.setAdapter(_habitItemAdapter);
+
+        /* Implementation of swipe menu functionality came from this source:
+        Name: Velmurugan
+        Date: March 4, 2021
+        URL: https://howtodoandroid.com/android-recyclerview-swipe-menu
+         */
+        // create a touch listener which handles the click and swipe function of the RecyclerView
+        RecyclerTouchListener touchListener = new RecyclerTouchListener(getActivity(), recyclerView);
+        touchListener.setClickable(new RecyclerTouchListener.OnRowClickListener() {
+            @Override
+            // if row at the specified position is clicked
+            public void onRowClicked(int position) {
+                openViewWindowForItem(position);
+            }
+        })
+                .setSwipeOptionViews(R.id.home_add_event_button)
+                .setSwipeable(R.id.habit_view, R.id.swipe_options, new RecyclerTouchListener.OnSwipeOptionsClickListener() {
+                    @Override
+                    public void onSwipeOptionClicked(int viewID, int position) {
+                        // open the habit event dialog
+                        openHabitEventsView(position);
+                    }
+                });
```

Comment:
New swipe button is here, and I'll probably need to change this to fit in with the new listener abstractions and such.